library/kani/src/lib.rs

```diff
@@ -95,10 +96,26 @@ pub fn any<T: Arbitrary>() -> T {
 /// kani::assume(char::from_u32(c as u32).is_ok());
 /// ```
 ///
+#[inline(never)]
+pub unsafe fn any_raw<T>() -> T
+where
+    // This where bound guarantees constant `size_of::<T>()` so we can use `size_of::<T>()`
+    // as a const generic argument and as a compile-time constant array length.
+    [(); std::mem::size_of::<T>()]:,
+{
+    let non_det_byte_arr = any_raw_inner::<{ std::mem::size_of::<T>() }>();
+    // We need `transmute_copy` instead of `transmute` because right now, rustc can't guarantee that the
+    // source and destination types are the same size, even though they are.
+    let non_det_var =
+        std::mem::transmute_copy::<[u8; std::mem::size_of::<T>()], T>(&non_det_byte_arr);
```

Comment:
Could there be issues with alignment here? If I understand correctly, the alignment of an array of `u8` is the same as `u8`. So could there be an issue if we transmute to a type (say `u64`) with a bigger alignment?

Reply:
According to https://doc.rust-lang.org/stable/std/mem/fn.transmute.html: 
> Because transmute is a by-value operation, alignment of the transmuted values themselves is not a concern. As with any other function, the compiler already ensures both T and U are properly aligned. However, when transmuting values that point elsewhere (such as pointers, references, boxes…), the caller has to ensure proper alignment of the pointed-to values.

Since both types don't have any pointers, I think we should be fine.